File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string my_string)
{
    int answer = 0;
    int A = 0;

    for (int i = 0; i < my_string.size(); i++)
    {
        if (my_string[i] >= '0' && my_string[i] <= '9')
            A = A * 10 + my_string[i] - 48;
        else
            answer += A, A = 0;
    }

    if (A != 0)
        answer += A;

    return answer;
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        
        char[] chars = my_string.toCharArray();
        int idx = chars.length - 1;
        
        while (idx >= 0) {
            int sum = 0;
            int d = 1;
            
            while (idx >= 0 && Character.isDigit(chars[idx])) {
                int digit = chars[idx] - '0';
                sum += digit * d;
                d *= 10;
                idx--;
            }
            
            answer += sum;
            idx--;
        }
        
        return answer;
    }
}